### PR TITLE
Add an Eliza-style chatbot.

### PR DESCRIPTION
**Pull request by freakboy3742**
This example is intended to demonstrate how to use the DetailedList widget to implement a chat interface.

### CODE REVIEW

File: examples/therapy/therapy/app.py [hunk 1]
```diff
@@ -33,6 +35,9 @@ class TherapyApp(toga.App):
             subtitle=response,
         )
 
+        # Scroll so the most recent entry is visible.
+        self.chat.scroll_to_row(len(self.chat.data) - 1)
```
Comment (goanpeca) [inline examples/therapy/therapy/app.py]: Now that I see this, should not we have some convenience method like

`scroll_to_last_row()`, `scroll_to_first_row()` or
`scroll_to_start()`, `scroll_to_end()`?

?

Comment (freakboy3742) [reply examples/therapy/therapy/app.py]: Makes sense - I'll add that when I get a chance.

File: src/core/toga/widgets/detailedlist.py [hunk 2]
```diff
@@ -58,6 +58,29 @@ def data(self, data):
         self._data.add_listener(self._impl)
         self._impl.change_source(source=self._data)
 
+    def scroll_to_top(self):
+        """Scroll the view so that the top of the list (first row) is visible
+        """
+        self._impl.scroll_to_row(0)
```
Comment (goanpeca) [inline src/core/toga/widgets/detailedlist.py]: Yeiiii, code review for the win XD!

**Comment by freakboy3742** [comment]
@Ocupe The perils of last minute refactoring :-) I've just pushed an update that should correct the problems you found - can you re-check?

**Comment by Ocupe** [comment]
@freakboy3742 now it works!

**Pull request merged**